internal/database/badger/database.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+// Package badger provides a database implementation using badger v3.
+package badger
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/internal/database"
+	"github.com/dgraph-io/badger/v3"
+)
+
+// Database is database implementation using a badger/v3 database.
+type Database struct {
+	badgerDatabase *badger.DB
```

Comment:
nit: embed?
```suggestion
	*badger.DB
```

Reply:
I prefer to avoid embedding if it's not needed because:

- it reduces method and field name conflicts
- it exports `DB` as a field accessible outside (big no no in that case)
- it's more cumbersome to assert against